tests/unit/test_winAPI/test_powerTracking.py

```diff
@@ -23,118 +24,118 @@ def setUp(self) -> None:
 			MagicMock(SystemPowerStatus())
 		)
 
-	def test_unknownPowerStatus_failedFetch(self):
+	def test_fetch_status_fetchFailed(self):

```

Comment:
is it worth splitting these test suites into contexts (fetch vs AC state change)?

Reply:
With a black box testing approach for `_powerTracking`, then to get good coverage you would want to test both contexts. White box testing allows us to know both contexts fetch the same information, and just format it differently. Ideally, the design would make this clear, and the two aspects (fetching the information, plugging it into a string format) could be tested independently.

One alternative might be to pass in a message formatter instead of an enum.
```python


class ReportPowerState:
    def powerStateMessage(powerChanged: bool, powerState: PowerState):
           text = []
           text.extend(_getBatteryInformation(systemPowerStatus))
           text.append(_getACStatusText(systemPowerStatus))
           return text

    def noBatteryMessage():
          return "
		return [_("No system battery")]

    def unknownMessage():
        return [_("Unknown power status")]

class ReportPowerStateChanged(ReportPowerState):
    def createMessage(powerChanged: bool, powerState: PowerState):
           text = []
           # Reverse the order since the change in battery percentage is more important.
           text.append(_getACStatusText(systemPowerStatus))
           text.extend(_getBatteryInformation(systemPowerStatus))
           return text

```

Reply:
This makes me wonder, should a change in AC state be reported differently? Using the logic: "the thing that changes should be dominant in the message reported", AC being unplugged (perhaps accidentally) should be highlighted.

Reply:
both contexts are currently being tested, the question was whether or not to separate the suites.

>should a change in AC state be reported differently? Using the logic: "the thing that changes should be dominant in the message reported", AC being unplugged (perhaps accidentally) should be highlighted.

This is the case currently

Reply:
>One alternative might be to pass in a message formatter instead of an enum.

Do you think this approach is required?

Reply:
Ok, we'll proceed as is.